src/theory/theory_engine.cpp

```diff
@@ -200,6 +201,13 @@ void TheoryEngine::finishInit()
     // finish initializing the theory
     t->finishInit();
   }
+
+  // Won't work with incremental solving, but the partitioning
+  // is focused on non-incremental anyway. 
```

Comment:
It would be good to add an options check to [this](https://github.com/cvc5/cvc5/blob/33fab4e47b9f5cc7dae76a30d7850a82bf8290d7/src/smt/set_defaults.cpp#L976) method.